In [64]:
from math import factorial

*Число* **сочетаний** из $n$ элементов по $k$ элементов в каждом (в сочетаниях порядок не важен):

$$C_n^k = \frac{n!}{k!(n - k)!}$$

**Напишем** функцию, чтобы вычислить количество сочетаний:

In [65]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### 1. Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

Найдем количество сочетаний крести из всех возможный крестей в 4 крести:

In [66]:
combinations(13, 4)

715

Остальные три карты выбираются из 39 карт  — крести не рассматриваются, так как они уже выбраны:

In [67]:
combinations(39, 0)

1

Р(вероятность) = число исходов, благоприятствующих нашему событию / общее число исходов

Общее число исходов это все возможные сочетания 4 карт из 52:

In [68]:
combinations(52, 4)

270725

In [69]:
# P = 
(715 + 1) / 270725

0.002644750207775418

**Ответ**: вероятность взять все карты крести из 4 карт в колоде из 52 карт - 0.0026 или около 0,3%

б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

**Чтобы** решить эту задачу, нужно сначала рассмотреть ситуацию, когда выбирается один туз из четырех.

Число таких сочетаний будет равно:

In [70]:
combinations(4, 1)

4

Остальные три карты выбираются из 32 карт  — тузы не рассматриваются, так как они уже выбраны:

In [85]:
combinations(48, 3)

17296

И в этом случае будет такое число сочетаний, когда из колоды, в которой 36 карт, выбраны четыре, одна из которых — туз:

In [89]:
4*17296

69184

Найдем по такому же принципу  число возможных сочетаний если у нас 2, 3 и 4 туза из 4:


In [86]:
(combinations(4, 2))*(combinations(48, 2))

6768

In [87]:
combinations(4, 3)*combinations(48, 1)

192

In [88]:
combinations(4, 4)*combinations(48, 0)

1

Тогда общее количество комбинаций где есть хотя бы один туз:

In [90]:
69184+6768+192+1

76145

Р(вероятность) = число исходов, благоприятствующих нашему событию / общее число исходов


Общее число исходов это все возможные сочетания 4 карт из 52:

In [77]:
combinations(52, 4)

270725

Р = 76145 / 270725

In [91]:
76145 / 270725

0.2812632745405855

**Ответ**: вероятность взять хотя бы 1 туз из 4 карт в колоде из 52 карт - 0.2812  или около 28%

### 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Всего возможный комбинаций у нас:

In [79]:
combinations(10, 3)

120

Правильный код при этом только 1, по этому:

In [80]:
# P = 
1/120

0.008333333333333333

**Ответ**: вероятность того, что человек не знающий код код откроет дверь с первой попытки - 0.008  или около 0,8%

### 3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

**Зависимыми** событиями называются, когда появление одного влияет на появление другого. Вероятность появления двух зависимых событий:
     # P(A*B) = P(A)* P(B|A)


In [81]:
# P = 
(9/15) * (8/14) * (7/13)

0.18461538461538457

**Ответ**: вероятность того, что все извлеченные детали окрашены - 0.1846  или около 18,5%

### 4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [82]:
# Вариант когда мы приобретаем 2 выигрышных билета это:
combinations(2, 2)

1

In [83]:
# Все возможные сочетания:
combinations(100, 2)

4950

In [84]:
1/4950

0.00020202020202020202

**Ответ**: вероятность того, что 2 приобретенных билета окажутся выигрышными - 0.0002  или около 0,02%